bulk_insert.go

```diff
@@ -32,7 +32,48 @@ func BulkInsert(db *gorm.DB, objects []interface{}, chunkSize int, excludeColumn
 	return nil
 }
 
+// BulkInsertWithAssigningIDs executes the query to insert multiple records at once.
+// it will scan 'returning id' to [returningId] after every insert.
+// it's necessary to set insert_option="returning id" in *gorm.DB
+//
+// [returningId] must be a *[]uint(for integer) or *[]string(for uuid)
+//
+// [objects] must be a slice of struct.
+//
+// [chunkSize] is a number of variables embedded in query. To prevent the error which occurs embedding a large number of variables at once
+// and exceeds the limit of prepared statement. Larger size normally leads to better performance, in most cases 2000 to 3000 is reasonable.
+//
+// [excludeColumns] is column names to exclude from insert.
+func BulkInsertWithAssigningIDs(db *gorm.DB, returningId interface{}, objects []interface{}, chunkSize int, excludeColumns ...string) error {
+	typ := reflect.TypeOf(returningId)
+	if typ.Kind() != reflect.Ptr || typ.Elem().Kind() != reflect.Slice {
+		return errors.New("returningId must be a slice ptr")
+	}
+
+	allIds := reflect.ValueOf(returningId).Elem()
+	typ = allIds.Type()
+
+	// Split records with specified size not to exceed Database parameter limit
+	for _, objSet := range splitObjects(objects, chunkSize) {
+		ids := reflect.New(typ)
+		scanReturningId := func(db *gorm.DB) error {
+			return db.Scan(ids.Interface()).Error
```

Comment:
`Scan` does not work as expected. I think we should use `Pluck` instead.

However, at the same time, we have to know the primary key of objects since not all table's primary key is named "id", so we need to examine `*gorm.Field` to know what is the primary key.

Reply:
yes, my playground is 
```
package main

import (
	"fmt"
	"time"

	"github.com/jinzhu/gorm"
	_ "github.com/jinzhu/gorm/dialects/postgres"
	gormbulk "github.com/t-tiger/gorm-bulk-insert"
)

type Model struct {
	Id        uint      `gorm:"primary_key"`
	CreatedAt time.Time `gorm:"default:now()"`
	Data      string
}

func main() {
	db, err := gorm.Open("postgres", "host=localhost port=5432 user=cloudwalker dbname=cloudwalker password=cloudwalker sslmode=disable")
	if err != nil {
		panic(err)
	}
	defer db.Close()
	db.SingularTable(true)

	if err := db.AutoMigrate(Model{}).Error; err != nil {
		panic(err)
	}

	models := []interface{}{
		Model{Data: "aaa"},
		Model{Data: "bbb"},
		Model{Data: "ccc"},
	}
	var returnModel []Model
	if err := gormbulk.BulkInsertWithAssigningIDs(
		db.Set("gorm:insert_option", "returning *"), &returnModel, models, 1000); err != nil {
		panic(err)
	}

	fmt.Printf("insert success with returning: %#v", returnModel)
}
```
it works well, but i miss a little bit of details.
IMO, we can assert ids(interface{}) as
- []struct: Scan ids
- []T: Pluck "primary key"

what do u think of?

Reply:
Yeah, we can assign struct values here. But I think we should separate methods for the case of receiving slice of struct and slice of IDs. Please take care of it next PR.